File: backend/app/models/portalitem.py
from ..db import db
from sqlalchemy import UUID

class Portalitem(db.Model):
    __tablename__ = 'portalitems'
    guid = db.Column(UUID, primary_key=True, nullable=False, unique=True)
    id = db.Column(db.String(120), nullable=False, unique=True)
    title = db.Column(db.String(255), nullable=True)
    type = db.Column(db.String(120), nullable=True)
    owner = db.Column(db.String(120), nullable=True)
    homepage = db.Column(db.String(255), nullable=True)
    url = db.Column(db.String(255), nullable=True)
    sizebyte = db.Column(db.Integer, nullable=True)
    sharing = db.Column(db.String(120), nullable=True)
    groupsharing = db.Column(db.String(255), nullable=True)
    folder = db.Column(db.String(255), nullable=True)
    editable = db.Column(db.Boolean, nullable=True)
    viewcount = db.Column(db.Integer, nullable=True)
    usagequota = db.Column(db.Integer, nullable=True)
    categories = db.Column(db.String(255), nullable=True)
    contentstatus = db.Column(db.String(120), nullable=True)
    createdat = db.Column(db.String(120), nullable=True)
    modifiedat = db.Column(db.String(120), nullable=True)
    snippet = db.Column(db.String(255), nullable=True)
    description = db.Column(db.String(500), nullable=True)
    thumbnail = db.Column(db.String(255), nullable=True)
    tags = db.Column(db.String(255), nullable=True)

    def to_dict(self):
        return {
            "guid": self.guid,
            "id": self.id,
            "title": self.title,
            "type": self.type,
            "owner": self.owner,
            "homepage": self.homepage,
            "url": self.url,
            "sizebyte": self.sizebyte,
            "sharing": self.sharing,
            "groupsharing": self.groupsharing,
            "folder": self.folder,
            "editable": self.editable,
            "viewcount": self.viewcount,
            "usagequota": self.usagequota,
            "categories": self.categories,
            "contentstatus": self.contentstatus,
            "createdat": self.createdat,
            "modifiedat": self.modifiedat,
            "snippet": self.snippet,
            "description": self.description,
            "thumbnail": self.thumbnail,
            "tags": self.tags
        }

    def __repr__(self):
        return f'Portal Item {self.title}'
    

File: frontend/src/models/portalitem.ts
import {AbstractModel} from "./abstract-models.ts";

interface PortalItemProps {
    guid: string;
    id: string;
    title?: string;
    type?: string;
    owner?: string;
    homepage?: string;
    url?: string;
    sizebyte?: number;
    sharing?: string;
    groupsharing?: string;
    folder?: string;
    editable?: boolean;
    viewcount?: number;
    usagequota?: number;
    categories?: string;
    contentstatus?: string;
    createdat?: string;
    modifiedat?: string;
    snippet?: string;
    description?: string;
    thumbnail?: string;
    tags?: string;
}

export class PortalItem extends AbstractModel {
    class: string = "PortalItem";

    guid: string;
    id: string;
    title?: string;
    type?: string;
    owner?: string;
    homepage?: string;
    url?: string;
    sizebyte?: number;
    sharing?: string;
    groupsharing?: string;
    folder?: string;
    editable?: boolean;
    viewcount?: number;
    usagequota?: number;
    categories?: string;
    contentstatus?: string;
    createdat?: string;
    modifiedat?: string;
    snippet?: string;
    description?: string;
    thumbnail?: string;
    tags?: string;

    constructor(props: PortalItemProps) {
        super()
        this.guid = props.guid;
        this.id = props.id;
        this.title = props.title;
        this.type = props.type;
        this.owner = props.owner;
        this.homepage = props.homepage;
        this.url = props.url;
        this.sizebyte = props.sizebyte;
        this.sharing = props.sharing;
        this.groupsharing = props.groupsharing;
        this.folder = props.folder;
        this.editable = props.editable;
        this.viewcount = props.viewcount;
        this.usagequota = props.usagequota;
        this.categories = props.categories;
        this.contentstatus = props.contentstatus;
        this.createdat = props.createdat;
        this.modifiedat = props.modifiedat;
        this.snippet = props.snippet;
        this.description = props.description;
        this.thumbnail = props.thumbnail;
        this.tags = props.tags;
    }
}